src/frame/frame.zig

```diff
@@ -301,12 +301,14 @@ pub fn Frame(comptime config: FrameConfig) type {
         pub const vector_length = config.vector_length;
 
         /// Returns the appropriate tail call modifier based on the target architecture.
-        /// WebAssembly doesn't support tail calls by default, so we use .auto for wasm targets.
+        /// WebAssembly and x86_64 stage2 backend don't support tail calls, so we use .auto for those targets.
         pub inline fn getTailCallModifier() std.builtin.CallModifier {
-            return if (builtin.target.cpu.arch == .wasm32 or builtin.target.cpu.arch == .wasm64)
+            return if (builtin.target.cpu.arch == .wasm32 or 
+                      builtin.target.cpu.arch == .wasm64 or
+                      builtin.target.cpu.arch == .x86_64)
```

Comment:
x86-64 is the main architecture we are targeting for tailcalls and is running on our macos machines